### DISCUSSION TITLE
Ideas to make onboarding easier for new users

### DISCUSSION THREAD

**Discussion by shafy** [mastodon/mastodon#20397] [General]
I think it's crucial to make the onboarding to Mastodon less confusing for new, non-tech-savvy users. Here are two problems and suggested solutions:

1) **When creating an account, it's confusing for users that they need to choose an instance**
We could mitigate this by suggesting mstdn.social as the default, preselected instance. We could show the option to select a different instances for users who want that. Now, I understand that this is a bit against the ethos of decentralized, but I think it's worth doing because a) we will get more users on Mastodon, and b) once they get the grasp of it, they can easily switch to another instance

2) **The distinction between local and federated timelines is confusing**
It's confusing because people are not used to it from other social media. I think we could improve this by making the federated and local menu entries in the right sidebar navigation less prominent. Would need to think through how exactly.

I'm happy to come up with a more detailed mock up for one or both of these issues if folks think it's worth pursuing.

Keep up the great work!

**Comment by btkostner** [comment]
One thing I've seen on the bird app was recommending changing the wording from instance to neighborhood to make onboarding more approachable by non technical users.

**Comment by dickhardt** [comment]
I explain to people that Mastodon is like email, where you pick your server and associated address. instance => server may be simpler for most people to grasp.

I'm opposed to mstdn.social, or any other specific server as a default. 

I do think having instances that do not require an application up at the top of the list to be a better choice. I have talked to a number of people that applied to be on an instance and never heard back (or did not see the acceptance).

**Comment by kawazoe** [comment]
I just invited a friend that I consider somewhat tech-savvy. They're not a dev, but know enough javascript and computers to install Windows on their own. Here's what they had to say about the on-boarding process (translated in english):

![image](https://user-images.githubusercontent.com/406575/208351508-f051cb38-977c-4f7a-81bb-cad232870960.png)

> This text is incredibly confusing. You've been talking about an "instance" but then, they use the word "server". At first, I wasn't sure if they were the same thing. What really threw me off was when I saw the word "network". To me, and I assume to most normies, servers and networks are the same thing! A server is a network of people. That's how it works with Discord! Also, why would I care where their account is hosted? It doesn't matter if they're in Europe or China...

So, there's a lot to unpack here. I think the most important thing to do is to standardize the lexicon around this stuff. For some reason, all of the official material from Mastodon uses the word "server", but the entire internet have been calling these "instances" for a while now! In fact, if I search for "find mastodon" on DDG, Brave, or Google, the autocomplete for "instance" comes before "server", and one of the best resources out there right now (instances.social) uses the word instance as well.

The one that really surprised me, however, was the distinction between network and server. I think this one could be fixed very easily by introducing people to the term "fediverse" right there instead of calling it "the network". It's a new word, so there is no risk of people having preconceptions about the nuance between the pipe and the endpoint. Having it as a blank slate here will make people curious about the concept. It will also make them more confortable seeing the federated timeline when they see it for the first time.

In fact, I think that something very easy could be done to improve this even further. The term server/instance and network/fediverse should be links or popovers that explains what they represent. Ideally, with a little drawing of a graph that highlights the different parts. I think this is even more important as the word "federated" gets thrown around a lot and no one around me knew what it meant and its implications. Reminder that I consider my friend in the top 15% of the population when it comes to tech-savviness, and they had trouble with these concepts, not because they are complicated, but because they are either new or ingrained with the wrong connotation.

This is also something that the joinmastodon.org page should do. The decentralized nature of the platform is the first thing that non-tech people are going to but their heads around. They should get a clear and concise idea of what it implies the second they look for mastodon or get invited by their friends. This would also fix the point that my friend mentioned. They are used to games like World of Warcraft, where if you join a server, your account is usually tied to a specific country. Even though they use Discord every day, they pictured Mastodon where servers represents physical locations and not communities. I suspect that this is because Mastodon servers, contrary to Discord servers, all have different addresses and loads in different pages. This is a contrast to where Discord servers are all in one place, both in term of addresses, but also all in the same app.

The official Android app, while better at explaining what "decentralized" means, also suffer from this issue. It's descriptions starts with "Mastodon is the largest decentralized social network on the internet. Instead of a single website, it's a network of million of users in independent communities that can all interact with one another, seamlessly." There is so much wrong with this description! It implies that each communities are equivalent to websites, which is kind-of true in the Mastodon world... except that it doesn't match the definition of "community" in the real world! You can already have multiple communities on the same site, or even communities across multiple sites, so why would Mastodon suggest that they are the same? Not only that, but it is also avoiding the term "server" and "instance" in this description, adding even more fuel in the confusion fire. It's also avoiding the word "federated" like the plague, which isn't a good thing considering how important the concept is to truly enjoy Mastodon as a whole.

**Reply by shafy** [reply]
I like these suggestions a lot! Does anybody know if official maintainers read these discussions?

**Reply by funficient** [reply]
Good suggestion!  There should definitely be a shared lexicon and a graph to explain the architecture will be very helpful! That's one of the biggest issues on the blockchain so we should prevent falling into the same pitfalls.   Maybe the docs already have something like that.

**Reply by funficient** [reply]
@shafy We need to create issues (look on the right navigation bottom) when we agree that it is something to fix.  But we'll need to split it out.  The changing of the wording suggested by @kawazoe can be one issue, while the getting started changes I suggested below will be another (maybe more than one) issue, and the login suggestion which I haven't added yet will be another.  

I posted on Mastadon for comment and will wait a bit.  If no-one objects or has any meaningful contribution, I will create an issue for the getting started part and let's then see what happens.

**Comment by funficient** [comment]


**Reply by shafy** [reply]
Hey, I think these are great ideas. Especially the onboarding ideas could simplify things.

**Reply by funficient** [reply]
Thank you.  I'm busy trying to do rough mockups and process flows to better illustrate what I mean, will post it once I'm done.

**Reply by funficient** [reply]
This is what I came up with for **getting started**.  
Since yesterday and today looks as if the twitter auth API isn't working anymore, but I don't think this will be an insurmountable problem. The principle would remain the same.  

Here is a link to a board with the screens and interaction.  You should be able to join as guest (just enter your name without registering) and view the details with comments, but also attaching some screendumps.

https://app.conceptboard.com/board/8fcu-gc90-rguk-ete6-9kyr

![image](https://user-images.githubusercontent.com/44545266/208419958-82a46094-8b5f-4ddb-b271-c4064104fbbb.png)
![image](https://user-images.githubusercontent.com/44545266/208420085-52f1d0f8-8967-4d89-b36d-6d7b042c9bf4.png)
![image](https://user-images.githubusercontent.com/44545266/208420154-1a8b3c9e-569b-478c-b607-b98bb69f138e.png)
![image](https://user-images.githubusercontent.com/44545266/208420222-8fed30aa-a467-4b47-b842-ca76ac9356d9.png)

**Reply by kawazoe** [reply]
I love the idea of not having to start fresh. I wish you didn't had to "authorize" twitter though. Who I follow on twitter is already public information, so I don't see why I shouldn't be able to just give it a username without having to sign in.

**Reply by shafy** [reply]
Thanks, this look cool! Regarding Twitter: Thinking about it, I'm not sure if it makes sense for Mastodon to focus so strongly on Twitter in the onboarding. There are already third party tools that do this, so I think it might be enough that Mastodon links in their docs somewhere to those tools. This is just from a branding / positioning point of view...

**Reply by funficient** [reply]
Linking to the docs via the front-end might be a good solution for the interim. 

The problem is that it's super confusing, complicated and overwhelming to get started and to find your friends on a fediverse. It's not as simple as signing up to Twitter, Facebook or Instagram and finding your friends, because you also have to know the server they're on.  

Currently it's impossible to find your friends unless you have their handle which I think is a barrier to entry.

**Reply by kawazoe** [reply]
@funficient I'm not convinced. Discord requires users to know a private random code to add them as friends - the part after the hash in your username. Those are completely random and that didn't slow them down or confused people that much. On there, I added most of my friends through servers we have in common and not through asking them their specific unique id.

The thing with social network is, people don't generally follow their close friends. I mostly follow people I have never met, with 2 or 3 exceptions, mostly because if we weren't connected, they would just constantly DM me on Telegram every single post they found interesting. I care about what they say and find their content engaging, and sometimes this is reciprocal and they follow me back. To me, it feels more like a club than a friend group. I go to mastodon.art to meet with my art club. I go to techhub.social to meet with my tech club. I go to mstdn.ca to meet with my Canadian fellows. Etc.

Mastodon also isn't Twitter. You cannot follow a thousand people here. You would have over 800 posts a day to scroll through, and that would take probably around 16 hours if you really engaged and socialized with everyone who posted them and their communities. Most people usually keep it at around 100 followers, and will unfollow old people as they follow more and their interests and social contacts shifts. If they followed someone because they care about a topic they're posting, they usually move over to following the relevant hashtag. If you try to follow everyone you followed on Twitter, you'll have a hard time. It's not meant to be a one place stop for #everything.

I had no trouble getting a lot of great people and bots to follow on Mastodon just with the search and hashtag features. If you want to follow your friends specifically, then maybe it would be worth getting together first and considering starting your own instance to move your accounts over there. You'll get a much better experience that way. In fact, this is often what happens on Discord too. They just made it very easy and straightforward to have your own place. This is why I said I love the idea of people not having to do a fresh start. Not because they should be able to follow everyone from their Twitter or Facebook, but because right now, it can be difficult to know if someone you really care about even is on Mastodon, or if you should throw them an invite, without asking them first directly (which you probably should anyway), or even if it might be worth starting your own place.

**Reply by funficient** [reply]
@kawazoe I was really hoping that I could contribute something valuable to the community as I've always wanted to be part of a community where I use the products I build too.  I see that is not possible.

There is no agreed vision or product direction so it's a bunch of conflicting opinions without someone to attempt to include different perspectives.  I should have realized it when I saw that the only valued contributions are coding or updating document, and not suggestions to shape the product.  My mistake. I take back all my suggestions.

**Reply by shafy** [reply]
@funficient I wouldn't take it personally. Code or doc updates are much easier for maintainers to agree on and merge, since it's much more "black and white" type of thing. I can imagine that it's very hard to manage a quickly growing open-source project like Mastodon, and I wouldn't be suprised if the maintainers get a lot of PRs and suggestions.

Now, I don't know who and how product decisions are taken on Mastodon, but it doesn't seem like the maintainers do it here on GitHub Discussions. I've heard someone else mention a Discord (I'm not in it). If you still have motivation, I'd join the Discord and ask the maintainers directly if and how you can provide UI/UX suggestions.

Again, please don't take the lack of action by the maintainers personally. I hope you keep contributing to Mastodon and many other open-source projects in the future.

Anyways, I wish you a Merry Christmas and Happy New Year 🎄🎁❤️

**Reply by funficient** [reply]
Thanks @shafy :-)  And Merry Christmas to you too!  I wouldn't say I'm demotivated, rather, realistic.  It's a developer driven product and they're doing a pretty good job actually.  

When they do have space for product and a more defined process for contributing (I reached out on three different channels to volunteer without any response, don't know about the Discord channel though) I would gladly contribute again, but right now, I feel it's not really productive (and I'm all about productivity).

Happy New Year!

**Comment by funficient** [comment]
I added https://github.com/mastodon/mastodon/issues/22506 as a suggestion just for the server page.  I also found a nice description of the fediverse in the documentation that can maybe be used to decide on terminology and add it there to standardize it.

**Comment by timtfj** [comment]
I joined a few weeks ago, and my main problem with the Federated timeline was finding a proper definition of what it is. There were a lot of confusing explanations, often from people who seemed confused about it themselves.

My current understanding is that the Federated timeline is simply:
* everyone on your server,
* and everyone they follow,
* but only the posts with Public visibilty.

So it's roughly like following everyone on your server plus everyone they follow. But I've also seen conflicting explanations which seem to define it as:

* everyone on your server,
* everyone they follow, 
* and everyone on the _those_ people's servers,
* but only the posts with Public visibility.

It was weeks before I finally settled on the first definition, and I'm still not 100% sure which is correct.

Explanations that I've seen often get bogged down in explaining what each server sees from where, or where an individual post will be seen, with lots of ". . . so if even one person on server A follows someone on server B, then . . . ", so they elaborate the concept rather than clarify it. Either that, or they use simple but vague language which doesn't give a clear picture.

I would have been helped early on by seeing a straightforward **definition** like the ones I've tried to give above. The current description that appears for the Federated feed mentions "people on other servers of the federated network that this server knows about", which I found vague and confusing.

**Reply by kawazoe** [reply]
Federated systems like the one Mastodon is made out of (the fediverse) are the third point of the triangle together with 100% public systems, and 100% private systems. A system that is 100% private cannot communicate with the outside. If Mastodon was like this, you would be the only one reading your posts - not very useful for a social network, but what you expect out of a word processing application. Systems that are 100% public ensures that every one have full access to the entire state of the system. This is like the blockchain in cryptocurrencies. Every participant needs to have a full copy of the thing on their computer, kept perfectly in sync, for it to work. Even if you're not very savvy, just thinking about the challenge of doing this should give you an idea of why: 1/ transactions doesn't scale very well in cryptoland, and 2/ how this would mean each Mastodon server would need the power of a whole data-center just to let you chat with a few of your friends.

Yet, somehow, Mastodon exists, and it lets you follow anyone on the fediverse, without the need for a billion dollar investment in Facebook-scale hardware just to get going. What gives? That's the third point of the triangle...

A Mastodon server is only aware of the data you give it, either directly by posting on that server, or through federated connections with other servers. But again, it cannot know of those connections unless you made them yourself in the first place. It's more like if you used Word to write your posts, but stored them on folder in Dropbox shared with your friends. They can view all of your posts, but they don't have the same friends as you, and so they can share it with their friends too, without asking you first, and you can see the post of your friends, but not necessarily the one their friends made. Each person in the chain have their own agency on whether they want to share posts with others, and vice versa. That's why they are called "federated". It's like a federation of states, each with their own internal policies and communities, but with standardized mechanisms to share information between the different states.

On Mastodon, it's up to you to *follow* people on other servers, and when you do, you tell your server to pull their data into your own server for your enjoyment, just like if you add that shared folder of your friend to your Dropbox. People have also compared it to subscribing to a newsletter for that person's timeline. Until you click that follow button, your server isn't aware of anything that person is posting. Following creates a connection between your server and their server that will propagate their timeline to your server every time they make a new post. This only happen for that person, not for their entire server, and not for the people they follow. You'll only get what they post or boost. Posts can also have replies from all over the place, so your server might be aware of those people too, just because they were mentioned, but it doesn't mean they will pull their entire timeline like for your follows.

The last point is that this person might not want all of their posts to be shared outside of their server, and so have different levels of reach they can chose when they post stuff, from DMs to local only all the way to the entire world. Stuff that's not public will never leave their local server.

With that in mind, the definition of the federated timeline becomes clear: It's everything your server is aware of. Same for the local timeline: It's everything that was posted directly to your server.

That whole thing is why I suggested that we explain the fediverse to people upfront instead of trying to hide it.

**Reply by timtfj** [reply]
From my point of view, that amounts to a long explanation of why my first definition is the correct one, I think.

There's a lot of detail there, but it seems to me quite a bit of it is about **how** federation is achieved—following someone causes your server to request information from the other server and store it, etc. From a new user's POV that's a separate issue from understanding 
**what** federation is—people on a whole load of different servers and types of server being able to communicate in a shared space that they mutually create.

My point is that the language needs to be as straightforward and specific as possible, and focused on what people need to know. New arrivals seem to get the impression that they have to learn lots of complicated technical stuff to even get started when actually they don't—all the concepts are straightforward ones.

IMO it's best to work out what the _minimum_ is that someone needs to know in order to find their way around, give them that in a very clear form so they can, and tell them where they can find more information if they want it. We don't learn by being hit with all the information at once—we learn by understanding basic concepts then building on them bit by bit.

I'd say the basic concepts here are:

• different users on different instances
• following people
• timelines and who ends up in the different ones
• visibility settings.

"People your server knows about" seems to be a less explicit way of saying "people on your server or followed by someone on it"—as far as I can tell, it defines exactly the same set of people—but it's less basic and needed explaining, so should be avoided in a basic introduction. (One reason it's confusing is that you can search for someone your server "doesn't know about", find them and follow them, all from your own server—which looks a lot like it knowing about them.)

Then maybe explain that, for example, certain types of information are only stored for people on or followed by someone on your server, and so on. But avoid unnecessarily multiplying concepts and analogies.

**Comment by trankten** [comment]
Mastodon has been growing a lot, but people still need to understand. I agree we need to fix the lexicon and the terms everywhere to be consistent... "Server", "Instance", "Host", "Network", "Fediverse"...

Sometimes two different terms are used in the same sentence to explain the same which is, indeed confusing.

Having a default choice breaks the freedom of Mastodon and Fediverse, so I really discourage this approach. I also discourage a "Find friends" integration to Twitter and/or any other social network has that means there's a centralized database storing all the users which are linked to a social network and that goes against decentralization too.

The right step would be to create a lexicon, easily understandable for new and existing users and expand that through the whole community to avoid confusion.

"Instance" is an awesome name but confusing for new people. "Server" is more easy to understand like "email server", and the comparison is quite similar to the usernames. Same goes for "Fediverse"... I'd use that instead "Mastodon Network" or just "Network". People who go deep into the word will understand that Fediverse goes to all ActivityPub related software but new people who read "Fediverse" the first time will understand that's as just a "network of Mastodon servers".

Have a nice day.
